wapitiCore/controller/wapiti.py

```diff
@@ -259,29 +259,29 @@ async def _init_report(self):
 
         for vul in vulnerabilities:
             self.report_gen.add_vulnerability_type(
-                vul.NAME,
-                vul.DESCRIPTION,
-                vul.SOLUTION,
-                flatten_references(vul.REFERENCES),
-                vul.WSTG_CODE
+                vul.name(),
+                vul.description(),
+                vul.solution(),
+                flatten_references(vul.references()),
+                vul.wstg_code()
```

Comment:
it is sad to use methods for this

Reply:
We need abstract method for type enforcing. Python allowed `@property` on top of `@classmethod` at some point but removed it in recent versions of Python.

That's why we have to use methods.